ceph-salt-formula/salt/_states/ceph_salt.py

```diff
@@ -33,13 +33,16 @@ def end_step(name):
     return _send_event('ceph-salt/step/end', data={'desc': name})
 
 
-def set_reboot_needed(name):
+def set_reboot_needed(name, force=False):
     ret = {'name': name, 'changes': {}, 'comment': '', 'result': False}
-    if __grains__.get('os_family') == 'Suse':
-        needs_reboot = __salt__['cmd.run_all']('zypper ps')['retcode'] > 0
+    if force:
+        needs_reboot = True
     else:
-        ret['comment'] = 'Unsupported distribution: Unable to check if reboot is needed'
-        return ret
+        if __grains__.get('os_family') == 'Suse':
+            needs_reboot = __salt__['cmd.run_all']('zypper ps')['retcode'] > 0
```

Comment:
Instead of checking the return code, we should check whether the output contains the string "No processes using deleted files found". If it contains this string, reboot is not needed. Otherwise, reboot is needed.

Reply:
> Instead of checking the return code, we should check whether the output contains the string "No processes using deleted files found". Reboot is needed in all cases **except** when the output contains this string.

Pushed a commit that will rely on `zypper ps` output instead of exit code.